xarray/coding/variables.py

```diff
@@ -159,30 +224,34 @@ def encode(self, variable: Variable, name: T_Name = None):
         fv = encoding.get("_FillValue")
         mv = encoding.get("missing_value")
 
-        if (
-            fv is not None
-            and mv is not None
-            and not duck_array_ops.allclose_or_equiv(fv, mv)
-        ):
-            raise ValueError(
-                f"Variable {name!r} has conflicting _FillValue ({fv}) and missing_value ({mv}). Cannot encode data."
-            )
+        if fv is not None or mv is not None:
```

Comment:
```suggestion
        fv_exists = fv is not None
        mv_exists = mv is not None
        if fv_exists or mv_exists:
```
A lot of the same checks over and over again, can make a variable instead and reuse that.
You can replace more below.

Reply:
Thanks I've already thought about how to make this more readable. Just out of interest @Illviljan, is this change more for readability or does it also have an impact on performance?

Reply:
Was mainly a micro-optimization to me, nothing major. From like 4x-"fx is not None" operations to 1. 
I just think it's good practice to avoid duplicating operations.

Reply:
I've applied your suggestion, @Illviljan together with @dcherian's.